data/tabular/train_test_split.py

```diff
@@ -0,0 +1,141 @@
+import tdc
+import pandas as pd
+import numpy as np
+from typing import List, Tuple, Dict, Union, Optional
+import sys
+from tqdm import tqdm
+from random import Random
+from collections import defaultdict
+from rdkit import Chem
+from rdkit.Chem.Scaffolds import MurckoScaffold
+from rdkit import RDLogger
+from glob import glob
+
+RDLogger.DisableLog("rdApp.*")
+
+def print_sys(s):
+    """system print
+
+    Args:
+        s (str): the string to print
+    """
+    print(s, flush=True, file=sys.stderr)
+
+
+def create_scaffold_split(df, seed, frac, entity):
+    """create scaffold split. it first generates molecular scaffold for each molecule and then split based on scaffolds
+    adapted from: https://github.com/mims-harvard/TDC/tdc/utils/split.py
+
+    Args:
+        df (pd.DataFrame): dataset dataframe
+        fold_seed (int): the random seed
+        frac (list): a list of train/valid/test fractions
+        entity (str): the column name for where molecule stores
```

Comment:
`entity` must be always a SMILES representation?

Reply:
i fixed this now